include/materials/texture.h

```diff
@@ -60,4 +60,21 @@ class image_texture : public texture {
     image image_data;
 };
 
+struct RGBA {
+  float A = 1.0;
+  color RGB = color(0,1,1);
+};
+// Specifically takes in PNGs and also accounts for alpha channel.
+class pixel_image_texture : public texture {
+  public:
+  pixel_image_texture(const char* filename);
+
+  color value(double u, double v, const point3& p) const; // should never be used, its simply purely virtual above
```

Comment:
if this should never be used, you could consider throwing an exception instead in the implementation file

Reply:
@connortbot honestly, we should just make all textures accept a color4 object imo, and then they just interact with a max transparency factor instead